src/lib/utils/fetchGitHubContributors.ts

```diff
@@ -0,0 +1,42 @@
+import { join } from "path"
+import { GITHUB_AUTH_HEADERS, GITHUB_COMMITS_URL } from "@/lib/constants"
+import type { Author } from "@/lib/interfaces"
+import type { FileContributorsState } from "@/lib/types"
+
+export const fetchGitHubAuthors = async (
+  relativePath: string
+): Promise<FileContributorsState> => {
+  const url = new URL(GITHUB_COMMITS_URL)
+  const CONTENT_DIR = "/src/content/" // TODO: Overriding with Gatsby repo—Remove and import from constants.ts for production
+  const filePath = join(CONTENT_DIR, relativePath, "index.md")
+  url.searchParams.set("path", filePath)
+
+  try {
+    const uniqueAuthors: Array<Author> = []
+    const response = await fetch(url, GITHUB_AUTH_HEADERS)
+    const commits = await response.json()
+    const authorSet = new Set<string>()
+    commits
+      .filter(({ author }) => author)
+      .forEach(({ author, commit }) => {
+        const entry: Author = {
+          name: commit.author.name,
+          email: commit.author.email,
+          avatarUrl: author.avatar_url,
+          user: {
+            login: author.login,
+            url: author.html_url,
+          },
+        }
+        // Unique authors only
+        authorSet.add(JSON.stringify(entry))
+      })
+    Array.from(authorSet).forEach((entry) =>
+      uniqueAuthors.push(JSON.parse(entry))
+    )
```

Comment:
maybe a simpler way could be
```suggestion
    const authorSet = Array.from(authorSet).map(JSON.parse)
```

Reply:
We would rename that variable, but I like this approach. TypeScript yelling at me when I try it though:

```ts
const authors = Array.from(authorSet).map(JSON.parse)
```
<img width="579" alt="image" src="https://github.com/ethereum/ethereum-org-fork/assets/54227730/c545df77-c8ae-46ee-95bc-c9659d1cab12">

This fixes it:

```ts
const authors = Array.from(authorSet).map(
  JSON.parse as (entry: string) => Author
)
```